sns_ik_lib/include/sns_ik/sns_ik_math_utils.hpp

```diff
@@ -25,66 +25,31 @@
 
 #include <Eigen/Dense>
 
-using namespace Eigen;
-
 namespace sns_ik {
 
-#define _USE_DOUBLE_
-/*! \def _USE_DOUBLE_
- * use values with double precision.
- * This is used to have the possibility to switch (at compile time) between float and double precision in order to be faster or more accurate.
- * Inside the \b IKL we will use the follow definition:
- * - \b MatrixD  a dynamic sized Matrix of real numbers
- * - \b VectorD a column vector of real numbers with dynamic length
- * - \b Scalar a real number
- */
-
-#ifdef _USE_DOUBLE_
-  /*! \typedef MatrixD
-   *  dynamic sized matrix of \em float values
-   *  (\em double if \b _USE_DOUBLE_ is defined)
-   */
-  typedef Eigen::Matrix<double, Dynamic, Dynamic> MatrixD;
-  /*! \typedef VectorD
-   *  dynamic sized column vector of \em float values
-   *  (\em double if \b _USE_DOUBLE_ is defined)
-   */
-  typedef Eigen::Matrix<double, Dynamic, 1> VectorD;
-  /*! \typedef Scalar
-   *  a \em float value
-   *  (\em double if \b _USE_DOUBLE_ is defined)
-   */
-  typedef double Scalar;
+namespace {
```

Comment:
Can you give a bit more color here? Is the anonymous namespace done in order to turn this API private?